### PR TITLE
Replace `str.removesuffix` usage in domain_record module

### PR DESCRIPTION
**Pull request by smkent**
`str.removesuffix` is new in Python 3.9:
https://docs.python.org/3/library/stdtypes.html#str.removesuffix

Python 3.7 and 3.8 are still in support:
https://devguide.python.org/versions/

On Python 3.7 or 3.8, the `domain_record` module fails with `failed to get domain record example.com: 'str' object has no attribute 'removesuffix'`

Python 3.8 is the default version of Python on Ubuntu 20.04 which is how this issue was discovered.

This change replaces the only usage of `str.removesuffix` in this repository with a conditional slice to restore compatibility with Python 3.7 and 3.8. This replacement provides the same functionality.

### CODE REVIEW

**Review by lgarber-akamai — APPROVED** [review]
Tested locally and all seems to work as intended. Thanks for the contribution!

Once this is merged, we can cut a patch release to unblock Python 3.7 and 3.8 users.

**Review by jriddle-linode — APPROVED** [review]

**Pull request merged**
